packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1843,6 +1845,7 @@ function renderRootSync(root: FiberRoot, lanes: Lanes) {
   const prevExecutionContext = executionContext;
   executionContext |= RenderContext;
   const prevDispatcher = pushDispatcher();
+  prepareRendererToRender(root.containerInfo);
```

Comment:
Instead of parallel, can this be inside the same `pushDispatcher()` function if we passed the root to it? Is there any place they diverge?

Reply:
yeah it could